### PR TITLE
PR review 

### PR DESCRIPTION
**Pull request by dmleong**
Draft review to start code review since there's no easy way to do a review without a PR

### CODE REVIEW

File: system/api-server/src/app.service.ts [hunk 1]
```diff
@@ -6,3 +6,4 @@ export class AppService {
     return 'Hello World!';
```
Comment (dmleong) [inline system/api-server/src/app.service.ts:6]: This feels like boilerplate, should we remove this?

Comment (Smeks-ops) [reply system/api-server/src/app.service.ts:6]: Yeah it's from the nestjs boilerplate so we can take it out

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ git clone https://github.com/Smeks-ops/maca-assessment.git
 
 ## Installing the App
 
-2. Install dependencies with this command in the home folder, the api-server folder and the client folder
+2. Install dependencies with this command in the home folder, the api-server folder and the client folder
```
Comment (dmleong) [inline README.md:15]: Nitpick: installation was a bit confusing here to have to run the same command in multiple directories

Comment (Smeks-ops) [reply README.md:15]: Ok yeah because I built it as a mono repo, I had three package.jsons

File: system/api-server/src/app.controller.ts [hunk 3]
```diff
@@ -10,3 +10,4 @@ export class AppController {
     return this.appService.getHello();
```
Comment (dmleong) [inline system/api-server/src/app.controller.ts:10]: Same here, boilerplate that should be removed

File: system/api-server/src/app.controller.spec.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ describe('AppController', () => {
 
   describe('root', () => {
     it('should return "Hello World!"', () => {
-      expect(appController.getHello()).toBe('Hello World!');
+      expect(appController.getHello()).toBe('Hello World!');
```
Comment (dmleong) [inline system/api-server/src/app.controller.spec.ts:19]: Remove as well

File: system/api-server/src/app.controller.ts (hunk 3 above)
Comment (Smeks-ops) [reply system/api-server/src/app.controller.ts:10]: Alright.

File: system/api-server/src/app.controller.spec.ts (hunk 4 above)
Comment (Smeks-ops) [reply system/api-server/src/app.controller.spec.ts:19]: ok

File: system/api-server/src/file-upload/file-upload.service.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ AWS.config.update({
   secretAccessKey: SECRET_ACCESS_KEY,
 });
 
-@Injectable()
+@Injectable()
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.service.ts:18]: This is a new type of behavior that I haven't seen in other languages. Can you explain to me what this does and what the benefit is?

File: system/api-server/src/file-upload/file-upload.service.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ AWS.config.update({
   secretAccessKey: SECRET_ACCESS_KEY,
 });
 
-@Injectable()
+@Injectable() 
 export class ImageUploadService {
   constructor(
     @InjectRepository(FileUpload)
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.service.ts:21]: Same thing here - why are we injecting a repository? Can you explain what this does?

File: system/api-server/src/file-upload/file-upload.service.ts [hunk 7]
```diff
@@ -33,7 +33,7 @@ export class ImageUploadService {
       Bucket: AWS_S3_BUCKET_NAME,
       Body: imageBuffer,
       Key: fileName,
-      ACL: 'public-read',
+      ACL: 'public-read',
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.service.ts:36]: Public read seems pretty permissive - are there other permissions you would like to explore here and how would you think about implementing it?

File: system/api-server/src/file-upload/file-upload.service.ts [hunk 8]
```diff
@@ -88,7 +88,7 @@ export class ImageUploadService {
     try {
       const v = await this.fileUploadRepository
       await v.find().then((data) => {
-        return res.status(201).json({ 
+        return res.status(200).json({
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.service.ts:91]: Wouldn't this be a 200 request since it's retrieving resources rather than signaling it's been created successfully?

File: system/api-server/src/file-upload/file-upload.controller.ts [hunk 9]
```diff
@@ -48,7 +48,7 @@ export class ImageUploadController {
         amount: {type: 'string', example: '100'},
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.controller.ts:48]: Should this be a decimal since it's an amount? Any type of number validation for currency?

File: system/api-server/src/file-upload/file-upload.service.ts (hunk 6 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.service.ts:21]: @Injectable() is how you tell the Nest framework that this is a class that can have dependencies that should be instantiated by Nest I think

File: system/api-server/src/file-upload/file-upload.controller.ts (hunk 9 above)
Comment (dmleong) [reply system/api-server/src/file-upload/file-upload.controller.ts:48]: Date should likely also not be a string and have validation

File: system/api-server/src/file-upload/file-upload.controller.ts [hunk 10]
```diff
@@ -81,7 +81,7 @@ export class ImageUploadController {
     @Res() response,
     @Query() params: GetUploadedFilesQuery,
   ) {
-    const { offset = 0, limit = 10 } = params;
+    const { offset = 0, limit = 10 } = params;
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.controller.ts:84]: Can you explain what this does? Is it to paginate? 

If so, it doesn't look like the app is paginating correctly

File: system/api-server/src/file-upload/file-upload.controller.ts [hunk 11]
```diff
@@ -58,7 +58,7 @@ export class ImageUploadController {
     @Res() response,
   ) {
     if (!file) {
-      throw new BadRequestException('invalid file provided');
+      throw new BadRequestException('Invalid file provided');
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.controller.ts:61]: Nitpick: for consistency's sake, use the same case for error messages

File: system/api-server/src/file-upload/file-upload.controller.ts [hunk 12]
```diff
@@ -38,7 +38,7 @@ export class ImageUploadController {
   @ApiBody({
```
Comment (dmleong) [inline system/api-server/src/file-upload/file-upload.controller.ts:38]: It'd be great to see some comments in these controllers to know what the app is doing

File: system/api-server/src/file-upload/file-upload.service.ts (hunk 7 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.service.ts:36]: I could make it private and use a signed URL to access the files. I haven't done this before though.

File: system/api-server/src/file-upload/file-upload.service.ts (hunk 8 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.service.ts:91]: Yes, it should be a 200 since the request was successful

File: system/api-server/src/file-upload/file-upload.controller.ts (hunk 9 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.controller.ts:48]: Actually, no currency validation was added here. I made it a string so that any value could be collected.

File: system/api-server/src/file-upload/file-upload.controller.ts (hunk 10 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.controller.ts:84]: It was to paginate. But then I wasn't sure how to paginate on the frontend correctly so I left it out

File: system/api-server/src/file-upload/file-upload.controller.ts (hunk 11 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.controller.ts:61]: Alright, I will. Thanks!

File: system/api-server/src/file-upload/file-upload.controller.ts (hunk 12 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.controller.ts:38]: Ok I will add comments to the controller and service

File: system/api-server/src/file-upload/file-upload.controller.ts (hunk 9 above)
Comment (Smeks-ops) [reply system/api-server/src/file-upload/file-upload.controller.ts:48]: I'll make changes

File: system/api-server/src/file-upload/entities/file-upload.entity.ts [hunk 13]
```diff
@@ -11,10 +11,10 @@ import {
 @Entity('file-upload')
 export class FileUpload {
   @PrimaryGeneratedColumn('uuid')
-  id: string;
+  id: string;
```
Comment (dmleong) [inline system/api-server/src/file-upload/entities/file-upload.entity.ts:14]: Shouldn't this be an int?

File: system/api-server/src/file-upload/entities/file-upload.entity.ts [hunk 14]
```diff
@@ -11,10 +11,10 @@ import {
 @Entity('file-upload')
 export class FileUpload {
   @PrimaryGeneratedColumn('uuid')
-  id: string;
+  id: string; 
 
   @Column({ type: 'varchar', length: 300, nullable: false })
-  date: string;
+  date: string;
```
Comment (dmleong) [inline system/api-server/src/file-upload/entities/file-upload.entity.ts:17]: Should this be a date rather than a string?

File: system/api-server/test/app.e2e-spec.ts [hunk 15]
```diff
@@ -19,6 +19,6 @@ describe('AppController (e2e)', () => {
     return request(app.getHttpServer())
       .get('/')
       .expect(200)
-      .expect('Hello World!');
+      .expect('Hello World!');
```
Comment (dmleong) [inline system/api-server/test/app.e2e-spec.ts:22]: Where are the rest of the tests..?

Comment (Smeks-ops) [reply system/api-server/test/app.e2e-spec.ts:22]: I initially thought there was a commit issue but after checking, I remembered I took them out because I had problems writing the tests for the create receipt endpoint because the upload part kept failing

File: system/client/src/react-app-env.d.ts [hunk 16]
```diff
@@ -1 +1,2 @@
 /// <reference types="react-scripts" />
```
Comment (dmleong) [inline system/client/src/react-app-env.d.ts:1]: Is this file needed? Looks like it's just commented code

File: system/client/src/index.tsx [hunk 17]
```diff
@@ -17,3 +17,4 @@ root.render(
 // to log results (for example: reportWebVitals(console.log))
 // or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
 reportWebVitals();
```
Comment (dmleong) [inline system/client/src/index.tsx:19]: This looks like boilerplate that should be removed

File: system/client/src/pages/UploadsPage/UploadsPage.tsx [hunk 18]
```diff
@@ -18,7 +18,7 @@ function UploadsPage() {
 
   const [data, setData] = useState<DataType[] | [] >([]);
 
-  // const data = [
+  // const data = [
```
Comment (dmleong) [inline system/client/src/pages/UploadsPage/UploadsPage.tsx:21]: This file is mostly commented code. This should be cleaned up before submitting a technical review

**Comment by dmleong** [comment]
Emailed results. Closing since no further action needed
